### DISCUSSION TITLE
A plugin tried to update a node field that it doesn't own:

### DISCUSSION THREAD

**Discussion by SamnnyWong** [gatsbyjs/gatsby#27039] [Help]
I am building a website base on this template: https://www.gatsbyjs.com/starters/netlify-templates/gatsby-starter-netlify-cms/
 This error occurred after I added new markdown files inside folders of pages.
The full error message is listed below:

ERROR #11321  PLUGIN

"gatsby-node.js" threw an error while running the onCreateNode lifecycle:

A plugin tried to update a node field that it doesn't own:

Node id: xxx
Plugin: default-site-plugin
name: slug
value: "All the new markdown files added by myself"

  79 |   if (node.internal.type === `MarkdownRemark`) {
  80 |     const value = createFilePath({ node, getNode })
> 81 |     createNodeField({
     |     ^
  82 |       node,
  83 |       name: `slug`,
  84 |       value,

File: gatsby-node.js:81:5

I am guessing I did something wrong when creating node in gastby-node.js, but I can't tell what's wrong, here is my set up:

```
exports.onCreateNode = ({ node, actions, getNode }) => {
  const { createNodeField } = actions
  fmImagesToRelative(node) // convert image paths for gatsby images

  if (node.internal.type === `MarkdownRemark`) {
    const value = createFilePath({ node, getNode })
    createNodeField({
      node,
      name: `slug`,
      value,
    })
  }
}
```

**Comment by pieh** [comment]
Are you able to share your code? You mentioned:
> I am building a website base on this template

But I don't know what "this template" is

**Comment by KyleAMathews** [comment]
What does `fmImagesToRelative` do?

**Comment by pieh** [comment]
> What does `fmImagesToRelative` do?

I assume that's helper function that previous version of https://github.com/danielmahon/gatsby-remark-relative-images#v2-breaking-changes plugin were exposing. It was causing problems because it was mutating node objects inline (but afaik ignored `internals` which is where field owners are stored right now)

**Comment by SamnnyWong** [comment]
> Are you able to share your code? You mentioned:
> 
> > I am building a website base on this template
> 
> But I don't know what "this template" is

Sorry I forget to put the template url in.
this is the template i am referring to: https://www.gatsbyjs.com/starters/netlify-templates/gatsby-starter-netlify-cms/
